src/algebra/classical_lie_algebras.lean

```diff
@@ -90,4 +130,195 @@ end
 
 end special_linear
 
+namespace symplectic
+
+/-- The matrix defining the canonical skew-symmetric bilinear form. -/
+def J : matrix (l ⊕ l) (l ⊕ l) R := matrix.from_blocks 0 (-1) 1 0
+
+/-- The symplectic Lie algebra: skew-adjoint matrices with respect to the canonical skew-symmetric
+bilinear form. -/
+def sp : lie_subalgebra R (matrix (l ⊕ l) (l ⊕ l) R) :=
+  skew_adjoint_matrices_lie_subalgebra (J l R)
+
+end symplectic
+
+namespace orthogonal
+
+/-- The definite orthogonal Lie subalgebra: skew-adjoint matrices with respect to the symmetric
+bilinear form defined by the identity matrix. -/
+def so : lie_subalgebra R (matrix n n R) :=
+  skew_adjoint_matrices_lie_subalgebra (1 : matrix n n R)
+
+/-- The indefinite diagonal matrix with `p` 1s and `q` -1s. -/
+def indefinite_diagonal : matrix (p ⊕ q) (p ⊕ q) R :=
+  matrix.diagonal $ sum.elim (λ _, 1) (λ _, -1)
+
+/-- The indefinite orthogonal Lie subalgebra: skew-adjoint matrices with respect to the symmetric
+bilinear form defined by the indefinite diagonal matrix. -/
+def so' : lie_subalgebra R (matrix (p ⊕ q) (p ⊕ q) R) :=
+  skew_adjoint_matrices_lie_subalgebra $ indefinite_diagonal p q R
+
+/-- A matrix for transforming the indefinite diagonal bilinear form into the definite one, provided
+the parameter `i` is a square root of -1. -/
+def Pso (i : R) : matrix (p ⊕ q) (p ⊕ q) R :=
```

Comment:
Why this name?

Reply:
Agreed, this is a poor name.

I was trying to go for something short and have a habit of calling these "change of basis" -type matrices `P` on paper and seem to have temporarily persuaded myself this would be OK. I use three such matrices in these changes with suffixes to distinguish:
  * `Pso` (here)
  * `PB` (lower down)
  * `PD` (lower down)

I agree we can do better though. Perhaps I should give up on the idea of brevity and replace these various `P`s with `cob_matrix` or something?